presto-main/src/main/java/com/facebook/presto/event/QueryMonitor.java

```diff
@@ -332,14 +333,21 @@ private static QueryIOMetadata getQueryIOMetadata(QueryInfo queryInfo)
 
     private static Map<String, String> mergeSessionAndCatalogProperties(SessionRepresentation session)
     {
-        ImmutableMap.Builder<String, String> mergedProperties = ImmutableMap.builder();
-        mergedProperties.putAll(session.getSystemProperties());
+        Map<String, String> mergedProperties = new LinkedHashMap<>(session.getSystemProperties());
```

Comment:
is it Linked for determinism and HashMap to allow overwrites?

Reply:
Yep.  Internally ImmutableMap is order preserving, and I figured I'd do the same just to make it easier for debugging.